### PR TITLE
Implement all

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

**Review by Max-Pochepets — APPROVED** [review]

**Review by AndriiBardin — APPROVED** [review]

File: src/main/java/com/mate/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        User secondUser = new User();
+        secondUser.setName("Alice");
+        secondUser.setLastName("Alyson");
+        userService.add(secondUser);
+        return "injected";
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable Long userId) {
+        return UserMapper.toDto(userService.get(userId));
```
Comment (Sofasmile) [inline src/main/java/com/mate/controller/UserController.java]: let's make UserMapper a component and remove static from the method

File: src/main/java/com/mate/dao/UserDao.java [hunk 2]
```diff
@@ -7,4 +7,6 @@
     void add(User user);
 
     List<User> getAll();
+
+    User get(Long id);
```
Comment (Sofasmile) [inline src/main/java/com/mate/dao/UserDao.java]:
```suggestion
    Optional<User> get(Long id);
```

File: src/main/java/com/mate/dao/UserDaoImpl.java [hunk 3]
```diff
@@ -45,4 +45,15 @@ public void add(User user) {
             throw new RuntimeException("User not found", e);
         }
     }
+
+    @Override
+    public User get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM User where id = :id", User.class)
```
Comment (Sofasmile) [inline src/main/java/com/mate/dao/UserDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_03.html

> Don’t complicate get by id methods in your DAO, use session.get().

File: src/main/java/com/mate/service/UserMapper.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+public class UserMapper {
+    public static User toObj(UserResponseDto dto) {
+        User user = new User();
+        user.setId(dto.getId());
+        user.setName(dto.getName());
+        user.setLastName(dto.getLastName());
+        return user;
+    }
+
+    public static UserResponseDto toDto(User user) {
```
Comment (Sofasmile) [inline src/main/java/com/mate/service/UserMapper.java]:
```suggestion
    public static UserResponseDto mapToDto(User user) {
```

File: src/main/java/com/mate/service/UserMapper.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.mate.service;
+
+import com.mate.dto.UserResponseDto;
+import com.mate.model.User;
+
+public class UserMapper {
+    public static User toObj(UserResponseDto dto) {
```
Comment (Sofasmile) [inline src/main/java/com/mate/service/UserMapper.java]: where do we use this method? Let's remove it)

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**